### PR TITLE
Dịch "weight-decay_vn" - Phần 4

### LINKED ISSUE #928: Dịch "weight-decay_vn" - Phần 4
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_multilayer-perceptrons/weight-decay_vn.md#L204-L308

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by tamnguyenvan**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #928

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [x] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [x] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 1]
```diff
@@ -207,25 +207,25 @@ test_iter = d2l.load_array(test_data, batch_size, is_train=False)
 ## Implementation from Scratch
 -->
 
-## *dịch tiêu đề phía trên*
+## Triển khai từ đầu
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]: fyi #812 
```suggestion
## Lập trình từ đầu
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 2]
```diff
@@ -207,25 +207,25 @@ test_iter = d2l.load_array(test_data, batch_size, is_train=False)
...
 
-### *dịch tiêu đề phía trên*
+### Khởi tạo tham số mô hình
 
 <!--
 First, we will define a function to randomly initialize our model parameters and run `attach_grad` on each to allocate memory for the gradients we will calculate.
 -->
 
-*dịch đoạn phía trên*
+Đầu tiên, chúng ta khai báo một hàm số để khởi tạo tham số cho mô hình một cách ngẫu nhiên và chạy `attach_grad` với mỗi phần bộ nhớ cấp phát cho gradient mà ta sẽ tính toán.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Đầu tiên, chúng ta khai báo một hàm số để khởi tạo tham số cho mô hình một cách ngẫu nhiên và chạy `attach_grad` với mỗi tham số để cấp phát bộ nhớ cho gradient mà ta sẽ tính toán.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 3]
```diff
@@ -207,25 +207,25 @@ test_iter = d2l.load_array(test_data, batch_size, is_train=False)
...
 
-## *dịch tiêu đề phía trên*
+## Triển khai từ đầu
 
 <!--
 Next, we will implement weight decay from scratch, simply by adding the squared $\ell_2$ penalty to the original target function.
 -->
 
-*dịch đoạn phía trên*
+Tiếp theo, chúng ta sẽ triển khai phân rã trọng số từ đầu, chỉ đơn giản bằng cách cộng thêm bình phương lượng xử phạt $\ell_2$ vào hàm mục tiêu ban đầu.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]: từ `lượng xử phạt` nghe hơi cồng kềnh nhỉ, để mình lên slack hỏi xem
```suggestion
Tiếp theo, chúng ta sẽ lập trình phân rã trọng số từ đầu, chỉ đơn giản bằng cách cộng thêm bình phương lượng xử phạt $\ell_2$ vào hàm mục tiêu ban đầu.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 4]
```diff
@@ -240,14 +240,15 @@ def init_params():
...
-### *dịch tiêu đề phía trên*
+### Khai báo Xử phạt Chuẩn $\ell_2$
 
 <!--
 Perhaps the most convenient way to implement this penalty is to square all terms in place and sum them up.
 We divide by $2$ by convention, (when we take the derivative of a quadratic function, the $2$ and $1/2$ cancel out, ensuring that the expression for the update looks nice and simple).
 -->
 
-*dịch đoạn phía trên*
+Có lẽ cách thuận tiện nhất để triển khai xử phạt này là bình phương tất cả các phần tử ngay tại chỗ và cộng dồn chúng lại.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Có lẽ cách thuận tiện nhất để lập trình xử phạt này là bình phương tất cả các phần tử ngay tại chỗ và cộng chúng lại với nhau.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 5]
```diff
@@ -240,14 +240,15 @@ def init_params():
...
+Ta đem chia $2$ cho tiện, (khi ta tính đạo hàm của hàm bậc hai, sô hạng $2$ và $1/2$ loại trừ nhau, đảm bảo biểu thức cập nhật trông dễ nhìn và gọn).
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]: tác giả thêm dấu má linh tinh vãi :))
```suggestion
Ta đem chia $2$ cho tiện (khi ta tính đạo hàm của hàm bậc hai, $2$ và $1/2$ sẽ loại trừ nhau, đảm bảo biểu thức cập nhật trông dễ nhìn và gọn).
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 6]
```diff
@@ -258,15 +259,17 @@ def l2_penalty(w):
...
 
 <!--
 The following code fits a model on the test set and evaluates it on the test set.
 The linear network and the squared loss have not changed since the previous chapter, so we will just import them via `d2l.linreg` and `d2l.squared_loss`.
 The only change here is that our loss now includes the penalty term.
 -->
 
-*dịch đoạn phía trên*
+Đoạn mã nguồn sau thực hiện khớp mô hình trên tập huấn luyện và thẩm định trên tập kiểm tra.
+Mạng tuyến tính và lỗi bình phương không thay đổi gì so với mục trước, vì vậy ta chỉ cần nhập chúng từ `d2l.linreg` và `d2l.squared_loss`.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Mạng tuyến tính và lỗi bình phương không thay đổi gì so với chương trước, vì vậy ta chỉ cần nhập chúng từ `d2l.linreg` và `d2l.squared_loss`.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 7]
```diff
@@ -258,15 +259,17 @@ def l2_penalty(w):
...
+Thay đổi duy nhất ở đây là hàm lỗi của ta giờ cả thành phần xử phạt.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Thay đổi duy nhất ở đây là hàm mất mát có thêm thành phần xử phạt.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 8]
```diff
@@ -292,14 +295,15 @@ def train(lambd):
...
+### Huấn luyện không có Điều chuẩn
 
 <!--
 We now run this code with `lambd = 0`, disabling weight decay.
 Note that we overfit badly, decreasing the training error but not the test error---a textook case of overfitting.
 -->
 
-*dịch đoạn phía trên*
+Giờ chúng ta chạy đoạn mã này với `lambd = 0`, vô hiệu hóa phân rã trọng số.
+Lưu ý rằng ta sẽ gặp hiện tượng quá khớp nặng, lỗi huấn luyện giảm nhưng lỗi kiểm tra không giảm---một trường hợp điển hình của hiện tượng quá khớp.
```
Comment (minhduc0711) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Hãy chú ý hiện tượng quá khớp nặng, lỗi huấn luyện giảm nhưng lỗi kiểm tra thì không---một trường hợp điển hình của hiện tượng quá khớp.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 9]
```diff
@@ -207,25 +207,25 @@ test_iter = d2l.load_array(test_data, batch_size, is_train=False)
...
 
-*dịch đoạn phía trên*
+Tiếp theo, chúng ta sẽ triển khai phân rã trọng số từ đầu, chỉ đơn giản bằng cách cộng thêm bình phương lượng xử phạt $\ell_2$ vào hàm mục tiêu ban đầu.
 
 <!--
 ### Initializing Model Parameters
 -->
 
-### *dịch tiêu đề phía trên*
+### Khởi tạo tham số mô hình
```
Comment (tiepvupsu) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
### Khởi tạo Tham số Mô hình
```

**Comment by tiepvupsu** [comment]
Khi nào có thời gian lên hoàn thành PR này nhé @tamnguyenvan

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 10]
```diff
@@ -292,14 +295,15 @@ def train(lambd):
 ### Training without Regularization
 -->
 
-### *dịch tiêu đề phía trên*
+### Huấn luyện không có Điều chuẩn
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
### Huấn luyện không Điều chuẩn
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 11]
```diff
@@ -207,25 +207,25 @@ test_iter = d2l.load_array(test_data, batch_size, is_train=False)
...
 
-## *dịch tiêu đề phía trên*
+## Lập trình từ đầu
 
 <!--
 Next, we will implement weight decay from scratch, simply by adding the squared $\ell_2$ penalty to the original target function.
 -->
 
-*dịch đoạn phía trên*
+Tiếp theo, chúng ta sẽ lập trình phân rã trọng số từ đầu, chỉ đơn giản bằng cách cộng thêm bình phương lượng xử phạt $\ell_2$ vào hàm mục tiêu ban đầu.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]: Từ `weight decay` đang được thảo luận ở #959; cần chốt từ trước khi merge PR này.

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 12]
```diff
@@ -240,14 +240,15 @@ def init_params():
 ### Defining $\ell_2$ Norm Penalty
 -->
 
-### *dịch tiêu đề phía trên*
+### Khai báo Xử phạt Chuẩn $\ell_2$
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]: Chỗ `pernalty` này cũng đang thấy thảo luận trên Slack. Cần thiết thì bổ sung vào Bảng thuật ngữ nghen (cc @minhduc0711)

Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
### Xác định Lượng phạt Chuẩn $\ell_2$
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 13]
```diff
@@ -240,14 +240,15 @@ def init_params():
...
 
 <!--
 Perhaps the most convenient way to implement this penalty is to square all terms in place and sum them up.
 We divide by $2$ by convention, (when we take the derivative of a quadratic function, the $2$ and $1/2$ cancel out, ensuring that the expression for the update looks nice and simple).
 -->
 
-*dịch đoạn phía trên*
+Có lẽ cách thuận tiện nhất để lập trình xử phạt này là bình phương tất cả các phần tử ngay tại chỗ và cộng chúng lại với nhau.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Có lẽ cách thuận tiện nhất để lập trình lượng phạt này là bình phương tất cả các phần tử ngay tại chỗ và cộng chúng lại với nhau.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 14]
```diff
@@ -240,14 +240,15 @@ def init_params():
...
+Ta đem chia $2$ cho tiện (khi ta tính đạo hàm của hàm bậc hai, $2$ và $1/2$ sẽ loại trừ nhau, đảm bảo biểu thức cập nhật trông dễ nhìn và gọn).
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Ta đem chia $2$ theo quy ước (khi ta tính đạo hàm của hàm bậc hai, $2$ và $1/2$ sẽ loại trừ nhau, đảm bảo biểu thức cập nhật trông đơn giản, dễ nhìn).
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 15]
```diff
@@ -258,15 +259,17 @@ def l2_penalty(w):
...
 <!--
 The following code fits a model on the test set and evaluates it on the test set.
 The linear network and the squared loss have not changed since the previous chapter, so we will just import them via `d2l.linreg` and `d2l.squared_loss`.
 The only change here is that our loss now includes the penalty term.
 -->
 
-*dịch đoạn phía trên*
+Đoạn mã nguồn sau thực hiện khớp mô hình trên tập huấn luyện và thẩm định trên tập kiểm tra.
+Mạng tuyến tính và lỗi bình phương không thay đổi gì so với chương trước, vì vậy ta chỉ cần nhập chúng từ `d2l.linreg` và `d2l.squared_loss`.
+Thay đổi duy nhất ở đây là hàm mất mát có thêm thành phần xử phạt.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Thay đổi duy nhất ở đây là hàm mất mát có thêm lượng phạt.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 16]
```diff
@@ -240,14 +240,15 @@ def init_params():
 ### Defining $\ell_2$ Norm Penalty
 -->
 
-### *dịch tiêu đề phía trên*
+### Xác định Lượng phạt Chuẩn $\ell_2$
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
### Định nghĩa Lượng phạt Chuẩn $\ell_2$
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 17]
```diff
@@ -258,15 +259,17 @@ def l2_penalty(w):
 ### Defining the Train and Test Functions
 -->
 
-### *dịch tiêu đề phía trên*
+### Khai báo hàm Huấn luyện và Kiểm tra
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
### Định nghĩa hàm Huấn luyện và Kiểm tra
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md [hunk 18]
```diff
@@ -292,14 +295,15 @@ def train(lambd):
...
-### *dịch tiêu đề phía trên*
+### Huấn luyện không Điều chuẩn
 
 <!--
 We now run this code with `lambd = 0`, disabling weight decay.
 Note that we overfit badly, decreasing the training error but not the test error---a textook case of overfitting.
 -->
 
-*dịch đoạn phía trên*
+Giờ chúng ta chạy đoạn mã này với `lambd = 0`, vô hiệu hóa phân rã trọng số.
```
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Giờ chúng ta chạy đoạn mã này với `lambd = 0`, vô hiệu hóa suy giảm trọng số.
```

File: chapter_multilayer-perceptrons/weight-decay_vn.md (hunk 11 above)
Comment (duythanhvn) [inline chapter_multilayer-perceptrons/weight-decay_vn.md]:
```suggestion
Tiếp theo, chúng ta sẽ lập trình suy giảm trọng số từ đầu, chỉ đơn giản bằng cách cộng thêm bình phương lượng phạt $\ell_2$ vào hàm mục tiêu ban đầu.
```

**Comment by tamnguyenvan** [comment]
All done.
